algorithms/gr-bfsrh/csharp/BreadthFirst.cs

```diff
@@ -0,0 +1,96 @@
+using System.Collections.Generic;
+
+namespace Algorithms
+{
+    //Based on https://www.geeksforgeeks.org/shortest-distance-two-cells-matrix-grid/
+    //Binary-first search
+    public class BreadthFirst
```

Comment:
Codacy found an issue: [Add a 'protected' constructor or the 'static' keyword to the class declaration.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6319078)